.github/workflows/checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Pull Request Checks
+
+on:
+  pull_request:
+    branches: ['main']
```

Comment:
This should also run when there's a push on the default branch, e.g., like [here](https://github.com/elixir-cloud-aai/proTES/blob/d13bc9221b01a7be81e84f7329a93c490f6772a2/.github/workflows/checks.yaml#L3-L7). This is important because when checking on the PR it is not _quite_ the same as a push. It sometimes happens that one works and the other fails, so if we don't have that check, we might have a broken default branch and don't know about it!

Reply:
Make sense! Will create the new file for that as push on the default might trigger the deployment for the `dev` environment in future!

Reply:
No need for a new file, you can also have these `on` conditionals on a per job basis.